app/src/controllers/attachments/sagas.js

```diff
@@ -0,0 +1,81 @@
+import { takeLatest, call, put, all } from 'redux-saga/effects';
+import { URLS } from 'common/urls';
+import { fetch } from 'common/utils';
+
+import { showModalAction } from '../../controllers/modal';
+import {
+  GET_ATTACHMENT_HAR,
+  ATTACHMENT_HAR_FETCH_ERROR,
+  GET_ATTACHMENT_BINARY,
+  ATTACHMENT_BINARY_FETCH_ERROR,
+  GET_ATTACHMENT_IMAGE,
+  ATTACHMENT_IMAGE_FETCH_ERROR,
+} from './constants';
+
+function fetchImageData({ projectId, binaryId }) {
+  return new Promise(() => ({
+    image: URLS.getFileById(projectId, binaryId),
+  }));
+}
+
+function fetchRealData({ projectId, binaryId }) {
+  return fetch(URLS.getFileById(projectId, binaryId));
+}
+
+/* IMAGE */
+
+function* openImageModalsWorker({ data: { projectId, binaryId } }) {
+  try {
+    const data = yield call(fetchImageData, { projectId, binaryId });
+    yield put(showModalAction({ id: 'attachmentImageModal', data }));
+  } catch (error) {
+    yield put({ type: ATTACHMENT_IMAGE_FETCH_ERROR, error });
```

Comment:
1. Please use action creators.
2. `payload: error`. If you want to mark this action as an error action, just set `error: true` (this is a common practice for FSA: https://github.com/redux-utilities/flux-standard-action#actions)

Reply:
@DiscoElevator that `catch` isn't even relevant, so I'll swap it's internals to a simple error log. `ATTACHMENT_IMAGE_FETCH_ERROR` is never consumed.